### ISSUE TITLE
Get AccessToken

### ISSUE THREAD

**Issue by tschuege** [alexa-js/alexa-app#116]
Hello

I try to get the access token but it's always null:

**Request** (LaunchRequest sent via Alexa Tester)

```
{
  "version": "1.0",
  "session": {
    "new": true,
    "sessionId": "amzn1.echo-api.session.abeee1a7-aee0-41e6-8192-e6faaed9f5ef",
    "application": {
      "applicationId": "amzn1.ask.skill.f20763c7-00a0-4cd5-b2c3-db4fca81e949"
    },
    "attributes": {},
    "user": {
      "userId": "amzn1.account.AM3B227HF3FAM1B261HK7FFM3A2",
      "accessToken" : "anyAccessToken"
    }
  },
  "request": {
    "type": "LaunchRequest",
    "requestId": "amzn1.echo-api.request.9cdaa4db-f20e-4c58-8d01-c75322d6c423",
    "timestamp": "2015-05-13T12:34:56Z"
  }
}
```

**Code to get the access token where accessToken is always null**

```
app.launch(function (req, res) {
    var accessToken = req.sessionDetails.accessToken;
    if (accessToken === null) {
        //no token! display card and let user know they need to sign in
        res.card({
            type: "LinkAccount"
        }).say("Please link your account");
    } else {
        //has a token
        res.say("Welcome!").shouldEndSession(false).send();
    }
});
```

Is that an issue or am I doing anything wrong?
Thanks!

**Comment by dblock** [comment]
Could be a bug. Try to write a test for it to demonstrate?

**Comment by tschuege** [comment]
I wrote a simple test and figured out that the accessToken actually gets delivered as expected. 
But it does not work when using the "Alexa Tester" user interface. So I guess I have to hunt the bug there....

**Comment by dblock** [comment]
So does this mean this issue here can be closed and the problem is that the Alexa Tester (on the Amazon website?) isn't sending it?

**Comment by tschuege** [comment]
No. Im talking about the Alexa Tester that is part of Alexa-App. The UI
that allows you to build and send requests.

**Comment by dblock** [comment]
Got it. Leaving this as is. You should dig deeper :)

**Comment by tschuege** [comment]
I figured out how to set the accessToken:
when set in server.js's function preRequest via req.body.session.user.accessToken it is available in the app.

But now i'm struggeling with the applicationId. Amazon recommends to check the applicationId in the skill which I would like to do in the app.pre function that gets called on each request. 
But how can I set it on server.js so I can use the Alexa Tester?
I tried various version like req.applicationId or req.body.applicationId but non of them made it availabe in req.applicationId where it is when invoked via Lambda.

Thanks for your help!
